src/Tuura/Boolean.hs

```diff
@@ -0,0 +1,86 @@
+module Tuura.Boolean (
+  module Tuura.Boolean.Parser,
+  CNF, DNF, Literal (..),
+  convertToCNF, genConcepts,
+  simplifyCNF, simplifyDNF, convertCNFtoDNF) where
+
+import Tuura.Boolean.Parser
+import Data.List
+import Data.List.Extra
+import Data.Foldable
+import Data.Maybe
+import Data.Ord
+
+type CNF a = [[Literal a]]
+
+type DNF a = [[Literal a]]
+
+data Literal a = Literal { variable :: a, polarity :: Bool } deriving (Eq, Ord)
+
+convertToCNF :: Eq a => (Expr a) -> CNF a
+convertToCNF expr = cnf
+  where
+    vars = toList expr
+    values = mapM (const [False, True]) vars
+    fs = filter (\v -> not $ eval expr (\x -> getValue x vars v)) values
+    genVar val var = Literal var val
+    sim = simplifyCNF $ map (\v -> nub $ (map (\f -> (genVar (v !! f) (vars !! f))) [0..(length vars - 1)])) fs
+    cnf = map (map (\s -> Literal (variable s) (not $ polarity s))) sim
+
+genConcepts :: Bool -> [Literal String] -> String
+genConcepts v e
+    | (length e) == 1 = causes ++ op ++ effect
```

Comment:
You don't need parentheses here. Perhaps, run your code through `hlint` -- it will catch a lot of small things like this.